feat: use tabster restorer API (#28530)
microsoft/fluentui: packages/react-components/react-dialog/stories/Dialog/DialogTriggerOutsideDialog.md

> I using a `Dialog` without a `DialogTrigger`, it is up to the user to make sure that the focus is restored correctly

Reviewer: ```suggestion
When a `Dialog` is used without a `DialogTrigger`, it is up to the user to make sure that the focus is restored correctly
```
